x-pack/auditbeat/module/system/socket/events/udp_queue_rcv_skb.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build linux,386 linux,amd64
+
+package events
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/x-pack/auditbeat/module/system/socket/common"
+	"github.com/elastic/beats/v7/x-pack/auditbeat/tracing"
+	"golang.org/x/sys/unix"
+)
+
+func validIPv4Headers(ipHdr uint16, udpHdr uint16, data []byte) bool {
+	return ipHdr != 0 &&
+		int(ipHdr)+20 < len(data) &&
+		data[ipHdr]&0xF0 == 0x40 &&
+		udpHdr != 0 &&
+		int(udpHdr)+12 < len(data)
+}
+
+type UDPQueueRcvSkbCall struct {
+	Meta   tracing.Metadata                 `kprobe:"metadata"`
+	Socket uintptr                          `kprobe:"sock"`
+	Size   uint32                           `kprobe:"size"`
+	LAddr  uint32                           `kprobe:"laddr"`
+	LPort  uint16                           `kprobe:"lport"`
+	IPHdr  uint16                           `kprobe:"iphdr"`
+	UDPHdr uint16                           `kprobe:"udphdr"`
+	Base   uintptr                          `kprobe:"base"`
+	Packet [common.SkBuffDataDumpBytes]byte `kprobe:"packet,greedy"`
+}
+
+func (e *UDPQueueRcvSkbCall) Flow() *common.Flow {
+	var remote *common.Endpoint
+	if valid := validIPv4Headers(e.IPHdr, e.UDPHdr, e.Packet[:]); valid {
```

Comment:
This looks broken after the refactor. The second call to validIPv4Headers below should only happen if the first one returns false. It's a workaround for the case where e.IPHdr/e.UDPHdr are (the lowest 16bit) of pointers instead of indices inside e.Packet.